stor/base.py

```diff
@@ -86,8 +87,126 @@ def _has_incompatible_path_module(self, other):
         compatible path module for path operations"""
         return isinstance(other, Path) and other.path_module != self.path_module
 
-    copy = utils.copy
-    copytree = utils.copytree
+    def copy(self, dest, **swift_retry_options):
+        """Copies a source file to a destination file.
+
+        Note that this utility can be called from either OBS, posix, or
+        windows paths created with ``stor.Path``.
+
+        Args:
+            self (path|str): The source directory to copy from
+            dest (path|str): The destination file or directory.
+            swift_retry_options (dict): Optional retry arguments to use for swift
+                upload or download. View the
+                `swift module-level documentation <swiftretry>` for more
+                information on retry arguments
+
+        Examples:
+            Copying a swift file to a local path behaves as follows::
+
+                >>> import stor
+                >>> swift_p = 'swift://tenant/container/dir/file.txt'
+                >>> # file.txt will be copied to other_dir/other_file.txt
+                >>> stor.copy(swift_p, 'other_dir/other_file.txt')
+
+            Copying from a local path to swift behaves as follows::
+
+                >>> from stor import Path
+                >>> local_p = Path('my/local/file.txt')
+                >>> # File will be uploaded to swift://tenant/container/dir/my_file.txt
+                >>> local_p.copy('swift://tenant/container/dir/')
+
+            Because of the ambiguity in whether a remote target is a file or directory, copy()
+            will error on ambiguous paths.
+
+                >>> local_p.copy('swift://tenant/container/dir')
+                Traceback (most recent call last):
+                ...
+                ValueError: OBS destination must be file with extension or directory with slash
+        """
+        from stor.obs import OBSUploadObject
+        dest = Path(dest)
+
+        if utils.is_filesystem_path(dest):
+            dest.parent.makedirs_p()
+            shutil.copy(self, dest)
+        else:
+            dest_file = dest if not dest.endswith('/') else dest / self.name
+            dest_obj_name = Path(dest_file.parent.resource or '') / dest_file.name
+            upload_obj = OBSUploadObject(self, dest_obj_name)
+            dest_file._copy_upload([upload_obj],
+                                   **swift_retry_options)
+
+    def copytree(self, dest, use_manifest=False, headers=None,
```

Comment:
same point here - move to filesystem path

Reply:
Moved